PWG/EMCAL/EMCALbase/AliAnalysisTaskEmcalEmbeddingHelperData.cxx

```diff
@@ -0,0 +1,97 @@
+#include <vector>
+
+#include "AliAnalysisTaskEmcalEmbeddingHelper.h"
+#include "AliAnalysisTaskEmcalEmbeddingHelperData.h"
+#include "AliAODEvent.h"
+#include "AliESDEvent.h"
+#include "AliVAODHeader.h"
+#include "AliEmcalList.h"
+
+/// \cond CLASSIMP
+ClassImp(AliAnalysisTaskEmcalEmbeddingHelperData);
+/// \endcond
+
+AliAnalysisTaskEmcalEmbeddingHelperData::AliAnalysisTaskEmcalEmbeddingHelperData():
+  AliAnalysisTaskEmcalEmbeddingHelper(),
+  fHistManager(0),
+  fPtSelectionMin(),
+  fPtSelectionMax()
+{
+}
+
+
+AliAnalysisTaskEmcalEmbeddingHelperData::AliAnalysisTaskEmcalEmbeddingHelperData(const char *name):
+  AliAnalysisTaskEmcalEmbeddingHelper(name),
+  fHistManager(name),
+  fPtSelectionMin(),
+  fPtSelectionMax()
+{
+}
+
+AliAnalysisTaskEmcalEmbeddingHelperData::~AliAnalysisTaskEmcalEmbeddingHelperData()
+{
+}
+
+void AliAnalysisTaskEmcalEmbeddingHelperData::UserCreateOutputObjects() 
+{
+
+  // run function from embedding helper base class
+  AliAnalysisTaskEmcalEmbeddingHelper::UserCreateOutputObjects();
+
+  TString histName = "fHistPtSelection";
+  TString histTitle = "pT selection;Result;p_{T}^{lead}";
+  auto histPtSelection= fHistManager.CreateTH2(histName, histTitle, 2, 0, 2, 200, 0, 100);
+  histPtSelection->GetXaxis()->SetBinLabel(1,"Accepted");
+  histPtSelection->GetXaxis()->SetBinLabel(2,"Rejected");
+
+  // Add all histograms to output list
+  TIter next(fHistManager.GetListOfHistograms());
+  TObject* obj = 0;
+  while ((obj = next())) {
+    fOutput->Add(obj);
+  }
+
+  PostData(1, fOutput);
+
+
+}
+
+/**
+ * Override inherited function and add check on track with max pT.
+ * Rejects event if track within set pT range is not found
+ */
+
+Bool_t AliAnalysisTaskEmcalEmbeddingHelperData::CheckIsEmbeddedEventSelected()
+{
+
+
+  Int_t nTracks = fExternalEvent->GetNumberOfTracks();
+  Int_t nTracksInRange = 0;
+  Double_t ptMax = 0;
+  for(Int_t i=0;i<nTracks;i++) {
+    Double_t pt = fExternalEvent->GetTrack(i)->Pt();
+    if(pt>ptMax) ptMax = pt;
+    //now check set windows
+    Int_t npt = fPtSelectionMin.size()==fPtSelectionMax.size()?fPtSelectionMin.size():0;
+    for(Int_t j=0;j<npt;j++) {
+      if(pt>fPtSelectionMin.at(j) && pt<fPtSelectionMax.at(j)) { 
+        // found track within pt
+        nTracksInRange++;
+      }
+    }
+  }
+
+  TString histName = "fHistPtSelection";
+  if(nTracksInRange==0) {
+    fHistManager.FillTH2(histName, 1.5, ptMax );
+    return kFALSE;
+  }
+  else {
+    fHistManager.FillTH2(histName, 0.5, ptMax );
+  }
+
+
+  if(!AliAnalysisTaskEmcalEmbeddingHelper::CheckIsEmbeddedEventSelected()) return kFALSE;
```

Comment:
I haven't checked this closely, but I would suggest moving this to the top of this function. No need to check the tracks if the event isn't selected anyway